pkg/kubelet/util/pluginwatcher/README

```diff
@@ -1,26 +1,36 @@
+# Plugin Registration Service
+
 This folder contains a utility, pluginwatcher, for Kubelet to register
 different types of node-level plugins such as device plugins or CSI plugins.
 It discovers plugins by monitoring inotify events under the directory returned by
-kubelet.getPluginsDir(). Lets refer this directory as PluginsSockDir.
-For any discovered plugin, pluginwatcher issues Registration.GetInfo grpc call
-to get plugin type, name and supported service API versions. For any registered plugin type,
-pluginwatcher calls the registered callback function with the received plugin
-name, supported service API versions, and the full socket path. The Kubelet
-component that receives this callback can acknowledge or reject the plugin
-according to its own logic, and use the socket path to establish its service
-communication with any API version supported by the plugin.
+kubelet.getPluginsDir(). We will refer to this directory as PluginsDir.
 
-Here are the general rules that Kubelet plugin developers should follow:
-- Run as 'root' user. Currently creating socket under PluginsSockDir, a root owned directory, requires
-  plugin process to be running as 'root'.
+Plugins are expected to implement the gRPC registration service specified in
+pkg/kubelet/apis/pluginregistration/v*/api.proto.
+
+
+## gRPC Service Lifecycle
+
+For any discovered plugin, kubelet will issue a Registration.GetInfo gRPC call
+to get plugin type, name, endpoint and supported service API versions.
+
+Kubelet will then go through a plugin initialization phase where it will issue
+Plugin specific calls (e.g: DevicePlugin::GetDevicePluginOptions).
+
+Once Kubelet determines that it is ready to use your plugin it will issue a
+Registration.NotifyRegistrationStatus gRPC call.
+
+
+## gRPC Service Overview
 
-- Implements the Registration service specified in
-  pkg/kubelet/apis/pluginregistration/v*/api.proto.
+Here are the general rules that Kubelet plugin developers should follow:
+- Run as 'root' user. Currently creating socket under PluginsDir, a root owned
```

Comment:
Is that `Run plugin as root` ?

Reply:
Well your plugin should have the correct permissions to create and read any files in the plugin directory.